src/pynwb/base.py

```diff
@@ -286,25 +287,50 @@ def __init__(self, **kwargs):
         self.images = images
 
 
+class TimeSeriesReference(namedtuple('TimeSeriesReference', ['idx_start', 'count', 'timeseries'])):
+    """
+    Class used to represent data values of a :py:class:`~pynwb.base.TimeSeriesReferenceVectorData`
+    This is a ``namedtuple('TimeSeriesReference', ['idx_start', 'count', 'timeseries'])`` type.
+    """
+    pass
+
+
 @register_class('TimeSeriesReferenceVectorData', CORE_NAMESPACE)
 class TimeSeriesReferenceVectorData(VectorData):
```

Comment:
Can we override `add_row` and `append` to restrict the type of the input to either a TimeSeriesReference or a tuple with length 3 and the correct data types?

It is currently possible to do:
```python
v = pynwb.base.TimeSeriesReferenceVectorData(name='a', description='a')
v.add_row(1)
v.append(1)
```

Reply:
Closing to comment on the PR instead of the commit.

Reply:
Done in https://github.com/NeurodataWithoutBorders/pynwb/pull/1349/commits/a58921aca2401c9c809b00a7bec47bf30d977558